### ISSUE TITLE
Define a nice art style for the 2D assets.

### ISSUE THREAD

**Issue by djrm** [djrm/open-game-assets#1]
We will start by making some nice 2d vector assets, but first we need to define a cool art style, since this work is going to used as an official resource for godot engine demos, it has to follow the Godot's key principles and values (friendly, accessible, fun, modern, innovative, ...).

ping
@NathanLovato

**Comment by fry-** [comment]
I would avoid overly shiny gradients and shadows ([example](https://image.freepik.com/free-vector/sport-balls-and-other-games-vector-graphics_659594.jpg)). This always looks like old stuff from the last two decades.
Flat colors ( [example1](https://cdn.tutsplus.com/vector/uploads/legacy/tuts/000-2012/553_game_character/final.jpg), [example2](https://opengameart.org/sites/default/files/forum_attachments/tutorial_sample11_1.jpg), [example3](http://68.media.tumblr.com/86ccbdb2048399cace2c063e0f7642d6/tumblr_nmueywKh1Q1sh566ao1_1280.jpg) ) are very modern and can help with games which demand clear looks for better playability (compare [vegg's clear looks](https://gitlab.com/johannesg/vegg) to [the overly crowded looks of Geometry Wars](https://youtu.be/qDDIYHDk5mo)). But maybe flat colors cannot show off some of Godot's 2D graphic features, like 2D lightning? At the other hand, vector graphics might be in general limited in this regard.
Flat colors also exist with outlines ( [example](https://3.bp.blogspot.com/-iE7WBqFaw74/VtF4PBJNofI/AAAAAAAAGC4/Io2eETPyeBs/s1600/Skeleton_character_sheet.jpg) ). This might help to flesh out some parts of the object, which could be helpful for games, but are not that modern IMO.
Maybe also not very modern, but still very appealing (to me) are images with just discreet gradients ( [example](http://4.bp.blogspot.com/-U47emIDBBCM/UFqI7164bPI/AAAAAAAAApk/hWuN7kL4tqU/s1600/MoralesA_Vector_004.png)). These might always need outlines, so different objects don't blend into each other because of similar gradients on their borders. Like in the shown example, different outline widths emphasize the object's features differently (also seen in Mangas).

I think these are the available style-options for vector graphics.
I personally tend to the "flat without outlines" style currently.

**Comment by NathanLovato** [comment]
There's a lot more you can do with vector but I can't work on an art style out of context. In general, either a flat or a toon/cel-shaded look will be the easiest to work with. And the most accessible.

The core values are a bit hard to all cover to me. Conveying a `modern` and `innovative` and `friendly` and `fun` feel is really hard. You don't really do game art like graphic design. Instead I'd recommend that we work for a given project, give ourselves a concrete context to start with and refine the art from there.

How about we pick a specific demo or set to work on first so we have a clear project and list of assets to flesh out the art direction? We also have the Godot-chan and Godot-bot designs to work with. 

I think the small team behind Superpowers really got the flow right, even though they stopped development and the main artist used their own style. If you want an entire asset pack art to have a strong personality you need to build a coherent world, so that characters, backgrounds, props relate to one another. That's the one element the Kenney assets packs lacked to me: they're meant to be a lot more generic and reusable I'd say so they don't have too distinct traits.

![image](https://user-images.githubusercontent.com/12694995/36298690-98eb486a-12f9-11e8-8106-9731ea12f354.png)

If I recommend to work per-project it's also because I don't think we can have a single art direction. It's almost always the game's design that constrains the art and not the other way around. The rare exceptions being some art-driven or narrative games. But even then, it generally starts with the game.
Say you want to add puzzle or UI demos at some point, or you want a different scale and proportions to cover different features (e.g. animation, rigging and mesh deform in 3.1)... you can reuse the character designs but the visual language will necessarily be a bit different. There's also 3d and pixel art demos that'll need to have their own AD if they ever happen.

## Project specs

We need some technical specifications to work with too if we're working on the Godot demos. Even if we can scale vector, the art's proportions matter. How about targetting either HD-ready or HD-full for all projects?

I'm looking to redo the outdated short platformer series on the channel. There's also 2 platformer demos

## Character and world design

How about we write about and use a "world of Godot" as a base for these asset projects? We have 2 complementary characters to work with already, just gotta turn them into much simpler game sprites ;)
The robotic elements can help give a sense of modernity while the cute faces and manga-inspired style can help us bring a sense of fun and accessibility? 

![image](https://user-images.githubusercontent.com/12694995/36299517-a2f95172-12fd-11e8-97a0-46d50dc1803b.png)

All suggestions. What do you think? By the way I'm looking to redo the platformer tutorial series on gdquest. Would you mind starting with a side-scrolling view? It'd change me from the current long top-down project ^^

**Comment by djrm** [comment]
@NathanLovato i like the idea of 'world of Godot (engine)' and doing material around the characters we already have, also the characters are modern, so i guess the theme would be futuristic, sci-fi or something along those lines, also since you have more experience in art direction, i would say you take the lead designing the first elements of this, as you say for the platformer, we could of course help adding concepts and mockups or giving feedback (with examples), we could catch later on, when the workflow is more defined and more work is needed (as i told you, i am not an artist so im not familiar with art workflows, examples will be really appreciated, and i will be making some art really soon).

Also i would avoid doing super specific elements (we are not trying to make a full game), and would go for a more generic approach, the point of this work in my opinion is easy prototyping for developers doing demos, meaning devs can get nice free assets for fancier demos, instead of adding some programmer art. In this sense reusability is an important property of the assets. the reasoning for this is that developer should be able to work on new demos fast, for example the patreon campaign demos are going to start soon, so it would be nice if developer could use some generic assets to do the promised demos (making them look nice in the process).

Of course we can also have some nice game-type specific assets for more elaborate demos (like highlighted demos or main content), but the priority should be reusable assets for quick prototyping.

**Comment by djrm** [comment]
also for some contextual reference i would look for pending demos in the patreon campaing. that would give us a reference on what is needed and what the people wants to see.

maybe @reduz can give us the list of requested demos.

**Comment by NathanLovato** [comment]
> Also i would avoid doing super specific elements (we are not trying to make a full game), [...] In this sense reusability is an important property of the assets

Yes. It's just that in practice this is almost impossible. Game assets (especially 2d) are designed for a given screen size, mechanics, world proportions, art direction... you can create reusable base shapes or meshes that people can combine, but not generic game-ready assets that fit many genres. That's why Kenney has so many different packs. One for each type of game.

But we can create reusable assets in the modular sense:

![image](https://user-images.githubusercontent.com/12694995/36338934-a3f509d2-13ba-11e8-8e85-9494c428d2ae.png)

Like tilesets, objects that slot in and combine well together. Sprites that also work for both manual placement and procedural level generation.

The Kenney packs are certainly well-designed when it comes to re-usability but they still have a particular style. The assets had to be flat and not too contrasted. Characters and the environments don't have a strong personality.

For things like characters and monsters it's hard to make them both generic and appealing.

But don't worry, we're on the same page. A great thing with this project is we can not only start to produce the assets but also do a bit of level design, animation, prepare scenes... both as part of the demos and to cater for the needs of devs.

On a side-note, there's one type of kit we can make reusable and generic: a pure prototyping kit with a simple visual language. This is not so generic but for the course I worked on some assets like these, to put a bit of color in the videos:

![image](https://user-images.githubusercontent.com/12694995/36339010-f9c8ab10-13bb-11e8-95c8-4d477f37114f.png)

Generic assets like these, maybe inspired from your graphic design's style, would be really valuable I think.

**Comment by djrm** [comment]
i think i was more inclined to create a generic, reusable and easy to use by devs assets kit (prototyping kit), of course more elaborate assets and full showcase levels are also very desirable but harder to make and harder to just drop in some random feature demo, so yeah, i think we can start by making a prototyping kit that developers and users can get and use right away.

**Comment by NathanLovato** [comment]
It's going to be very simple then. Squares, rectangles...

![image](https://user-images.githubusercontent.com/12694995/36611781-d899d6da-18d4-11e8-8f07-1ee33fcbc02e.png)

What do you think of a light blueprint-like ground/tileset, vibrant shapes for interactive objects and either a dark or light color for the background? Gotta try this out

![800px_colourbox8128101](https://user-images.githubusercontent.com/12694995/36611665-830a7f26-18d4-11e8-9d5a-06691339f24e.jpg)

**Comment by djrm** [comment]
yeah, absolutely, i was thinking about something like your second image, with prototype like elements, maybe adding basic shapes, some generic robots and characters.

This could also make it easier to transform feature demos into interactive tutorials or something like that.
